src/registries/yarn-registry.js

```diff
@@ -56,14 +57,22 @@ export default class YarnRegistry extends NpmRegistry {
   homeConfig: Object;
 
   getOption(key: string): mixed {
-    let val = this.config[key] || this.registries.npm.getOption(npmMap[key]);
+    let val = this.config[key];
 
-    // if we have no yarn option for this or have used a default then use the npm
-    // value if it exists
-    if (!val || val === DEFAULTS[key]) {
-      val = this.registries.npm.getOption(key) || val;
+    // if this isn't set in a yarn config, then use npm
```

Comment:
Can you please remove this commit from the PR, happy to accept the other option but I have problems with this code that I'll address in your other PR.